src/game/ZoneArea.hxx

```diff
@@ -0,0 +1,165 @@
+   * 
+   * @param Coordinates of the mapNode with a zone tile
+   */
+  void addZoneNode(ZoneNode zoneNode);
+
+  /**
+   * @brief Remove a coordinates to a mapNode that has a zone placed on it.
+   * 
+   * @param Coordinates of the mapNode with a zone tile
+   */
+  void removeZoneNode(Point coordinate);
+
+  /**
+   * @brief Remove coordinates to a vacant mapNode (without a building) that has a zone placed on it.
+   * 
+   * @param Coordinates of the mapNode with a zone tile
+   */
+  void removeEmptyZoneNode(Point Coordinate);
+
+  /**
+   * @brief Set the power supply for this Area
+   * 
+   * @param whether the area has power supply or not
+   */
+  void setPowerSupply(bool hasPower) { m_hasPower = hasPower; };
+
+  /**
+   * @brief Get the Power supply for this Area
+   * 
+   * @return if the area has power supply
+   */
+  bool hasPowerSupply() { return m_hasPower; };
+
+  /**
+   * @brief Set the water supply for this Area
+   * 
+   * @param whether the area has water supply or not
+   */
+  void setWaterSupply(bool hasWater) { m_hasWater = hasWater; };
+
+  /**
+   * @brief Get the water supply for this Area
+   * 
+   * @return if the area has water supply
+   */
+  bool hasWaterSupply() { return m_hasWater; };
+
+  /**
+   * @brief Set the zone for this Area
+   * 
+   * @param zone for this area
+   */
+  void setZone(ZoneType zoneType) { m_zoneType = zoneType; };
+  ;
+
+  /**
+   * @brief Get the zone for this Area
+   * 
+   * @return zone for this area
+   */
+  ZoneType getZone() const { return m_zoneType; };
+
+  /**
+   * @brief Set the zone density for this Area
+   * 
+   * @param zone density for this area
+   */
+  void setZoneDensity(ZoneDensity zoneDensity) { m_zoneDensity = zoneDensity; };
+
+  /**
+   * @brief Get the zone density for this Area
+   * 
+   * @return zone density for this area
+   */
+  ZoneDensity getZoneDensity() { return m_zoneDensity; };
+
+  /**
+   * @brief If this coordinate is a neighbor of one of the tiles in this zone area.
+   * 
+   * @param coordinate The point to check
+   * @return neighbor of this zoneArea
+   */
+  bool isNeighborOfZone(Point coordinate) const;
+
+  /**
+   * @brief If this coordinate part of this zone area.
+   * 
+   * @param coordinate The point to check
+   * @return neighbor of this zoneArea
+   */
+  bool isPartOfZone(Point coordinate) const;
+
+    /**
+   * @brief If this area has unoccupied nodes left
+   * 
+   * @return neighbor of this zoneArea
+   */
+  bool isVacant() const;
+
+  /**
+   * @brief Spawn buildings on nodes in this area if all demands are fulfilled
+   * 
+   */
+  void spawnBuildings();
+
+  size_t getSize() { return m_zoneNodes.size(); };
+
+  bool isWithinBoundaries(Point coordinate) const
+  {
+    return (xmin <= coordinate.x) && (xmax >= coordinate.x) && (ymin <= coordinate.y) && (ymax >= coordinate.y);
+  }
+
+  void occupyZoneNode(Point coordinate);
+  void freeZoneNode(Point coordinate);
```

Comment:
Consider merging to one function setOccupancy.

Reply:
I called it setVacancy()